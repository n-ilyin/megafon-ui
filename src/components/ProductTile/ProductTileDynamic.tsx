import * as React from 'react';
import { cnCreate } from '../../utils/cn';
import './style/ProductTileDynamic.less';
import ProductSwitcher from '../ProductSwitcher/ProductSwitcher';
import AnimationValue from './ProductTileValue';
import { IProductTileDynamic } from './types';
import { productTileDynamicProps } from './propTypes';

const cn = cnCreate('mfui-product-tile-dynamic');
class ProductTileDynamic extends React.Component<IProductTileDynamic> {
    static propTypes = productTileDynamicProps;

    render() {
        const {
            currentPack,
            switcher,
            startCallsIndex,
            startTrafficIndex,
            onChangeTraffic,
            onChangeCalls,
        } = this.props;

        return (
            <div className={cn('')}>
                <div className={cn('constructor-pack')}>
                    <AnimationValue
                        hAlign="center"
                        value={`${currentPack.calls!.value} ${currentPack.calls!.unit}`}
                    />
                </div>
                <ProductSwitcher
                    className={cn('constructor-range')}
                    startIndex={startCallsIndex}
                    theme="tariff-showcase"
                    items={switcher.calls.map((value: string) => ({ title: value, value }))}
                    onChange={onChangeCalls}
                />
                <ProductSwitcher
                    className={cn('constructor-range')}
                    startIndex={startTrafficIndex}
                    theme="tariff-showcase"
                    items={switcher.traffic.map((value: string) => ({ title: value, value }))}
                    onChange={onChangeTraffic}
                />
                <div className={cn('constructor-pack')}>
                    <AnimationValue
                        hAlign="center"
                        value={`${currentPack.traffic!.value} ${currentPack.traffic!.unit}`}
                    />
                </div>
            </div>
        );
    }
}

export default ProductTileDynamic;
