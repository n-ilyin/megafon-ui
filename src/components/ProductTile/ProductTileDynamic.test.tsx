import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTileDynamic from './ProductTileDynamic';

const props = {
    currentPack: {
        calls: {
            value: 100,
            unit: 'минут',
        },
        traffic: {
            value: 10,
            unit: 'ГБ',
        },
    },
    switcher: {
        calls: ['1', '2'],
        traffic: ['3', '4'],
    },
    startCallsIndex: 1,
    startTrafficIndex: 1,
};

describe('<ProductTileDynamic />', () => {
    it('it renders ProductTileDynamic', () => {
        const wrapper = shallow(
            <ProductTileDynamic
                {...props}
                onChangeCalls={jest.fn()}
                onChangeTraffic={jest.fn()}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
