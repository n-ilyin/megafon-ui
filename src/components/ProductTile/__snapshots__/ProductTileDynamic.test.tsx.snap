// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTileDynamic /> renders ProductTileDynamic 1`] = `
<div
  className="mfui-product-tile-dynamic"
>
  <div
    className="mfui-product-tile-dynamic__constructor-pack"
  >
    <ProductTileValue
      hAlign="center"
      value="100 минут"
    />
  </div>
  <ProductSwitcher
    className="mfui-product-tile-dynamic__constructor-range"
    items={
      Array [
        Object {
          "title": "1",
          "value": "1",
        },
        Object {
          "title": "2",
          "value": "2",
        },
      ]
    }
    onChange={[MockFunction]}
    startIndex={1}
    theme="tariff-showcase"
  />
  <ProductSwitcher
    className="mfui-product-tile-dynamic__constructor-range"
    items={
      Array [
        Object {
          "title": "3",
          "value": "3",
        },
        Object {
          "title": "4",
          "value": "4",
        },
      ]
    }
    onChange={[MockFunction]}
    startIndex={1}
    theme="tariff-showcase"
  />
  <div
    className="mfui-product-tile-dynamic__constructor-pack"
  >
    <ProductTileValue
      hAlign="center"
      value="10 ГБ"
    />
  </div>
</div>
`;
